### PR TITLE
Exitcodes

### PR DESCRIPTION
**Pull request by rhatdan**
Handle exit codes properly.  According to the podman run Man page.

### CODE REVIEW

**Comment by rhatdan** [comment]
This patch also fixes a huge bug where if we failed to create a container, we left storage mounted.
@mheon @baude @TomSweeneyRedHat PTAL

File: libpod/container.go [hunk 1]
```diff
@@ -553,6 +553,13 @@ func (c *Container) Init() (err error) {
 	if err := c.mountStorage(); err != nil {
 		return err
 	}
+	defer func() {
+		if err != nil {
+			if err2 := c.cleanupStorage(); err2 != nil {
+				logrus.Errorf("Error tearing down network namespace for container %s: %v", c.ID(), err2)
```
Comment (mheon) [inline libpod/container.go]: Error message is wrong, this is tearing down storage not the network

**Comment by mheon** [comment]
LGTM

File: libpod/container.go (hunk 1 above)
Comment (rhatdan) [reply libpod/container.go]: Fixed

**Comment by baude** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
Changes LGTM, but there's all kinds of test unhappiness.

**Comment by rhatdan** [comment]
I think this is a flake.  Attempting to merge.

@rh-atomic-bot r=TomSweeneyRedHat

**Comment by rhatdan** [comment]
Finally passing tests.

@rh-atomic-bot r=TomSweeneyRedHat

**Pull request closed without merging**